src/vue/fields/DateField.vue

```diff
@@ -93,139 +118,228 @@ export default {
   },
 
   watch: {
-    'value': function () {
-      this.flatpickrDate = this.value
+    value (newValue) {
+      // Prevent updates if v-model value is same as input's current value
+      if (newValue === this.flatpickrDate) return
+      // Sets the current selected date after value changed
+      if (this.flatpickr) this.flatpickr.setDate(newValue, true)
+    },
+
+    config: {
+      deep: true,
+      handler (newConfig) {
+        let safeConfig = Object.assign({}, newConfig)
+
+        // Workaround: Don't pass hooks to configs again otherwise
+        // previously registered hooks will stop working
+        // This also means that new callbacks cannot be passed once a component
+        // has been initialized
+
+        this.flatpickr.set(this.removeFlatpickrHooks(safeConfig))
+      },
+    },
+
+    /**
+     * We can set disabled state only directly, because of flatpickr does not
+     * provide any config options
+     *
+     * @link https://github.com/flatpickr/flatpickr/issues/777
+     */
+    disabled (value) {
+      if (value) {
+        this.flatpickr._input.setAttribute('disabled', 'disabled')
+      } else {
+        this.flatpickr._input.setAttribute('disabled', '')
+      }
     },
   },
 
-  created () {
-    this.flatpickrDate = this.value
+  mounted () {
+    if (this.flatpickr) return
+
+    let safeConfig = Object.assign({}, this.config)
+
+    // Inject defined methods into events array
+    Object.values(BUILTIN_EVENTS).forEach(hook => {
```

Comment:
define an object with a method name and remove `BUILTIN_EVENTS`

Reply:
Looks shitty [commit](https://github.com/tokend/web-client/pull/238/commits/46f99ca6b02dfeb82c16985fb874b3a0685199e3)